server/notification-providers/pagerduty.js

```diff
@@ -0,0 +1,99 @@
+    async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
+        try {
+            if (heartbeatJSON == null) {
+                const title = "Uptime Kuma Alert";
+                const monitor = {
+                    type: "ping",
+                    url: "Uptime Kuma Test Button",
+                };
+                return await this.postNotification(notification, title, "", monitor);
+            }
+
+            if (heartbeatJSON.status === UP) {
+                const title = "Uptime Kuma Monitor ✅ Up";
+                const eventAction = notification.pagerdutyAutoResolve || null;
+
+                return await this.postNotification(notification, title, heartbeatJSON.msg, monitorJSON, eventAction);
+            }
+
+            if (heartbeatJSON.status === DOWN) {
+                const title = "Uptime Kuma Monitor 🔴 Down";
+                return await this.postNotification(notification, title, heartbeatJSON.msg, monitorJSON, "trigger");
+            }
+        } catch (error) {
+            this.throwGeneralAxiosError(error);
+        }
+    }
+
+    // thrown if failed to check result, result code should be in range 2xx
+    checkResult(result) {
+        if (result.status == null) {
+            throw new Error("PagerDuty notification failed with invalid response!");
+        }
+        if (result.status < 200 || result.status >= 300) {
+            throw new Error("PagerDuty notification failed with status code " + result.status);
+        }
+    }
+
+    async postNotification(notification, title, body, monitorInfo, eventAction = "trigger") {
+
+        if (eventAction == null) {
+            return "No action required";
+        }
+
+        let monitorUrl;
+        if (monitorInfo.type === "port") {
+            monitorUrl = monitorInfo.hostname;
+            if (monitorInfo.port) {
+                monitorUrl += ":" + monitorInfo.port;
+            }
+        } else if (monitorInfo.hostname != null) {
+            monitorUrl = monitorInfo.hostname;
+        } else {
+            monitorUrl = monitorInfo.url;
+        }
+
+        const options = {
+            method: "POST",
+            url: notification.pagerdutyIntegrationUrl,
+            headers: { "Content-Type": "application/json" },
+            data: {
+                payload: {
+                    summary: `[${title}] [${monitorInfo.name}] ${body}`,
+                    severity: notification.pagerdutyPriority || "warning",
+                    source: monitorUrl,
+                },
+                routing_key: notification.pagerdutyIntegrationKey,
+                event_action: eventAction,
+                dedup_key: "Uptime Kuma/" + monitorInfo.id,
+            }
+        };
+
+        const baseURL = await setting("primaryBaseURL");
+        if (baseURL && monitorInfo) {
+            options.client = "Uptime Kuma";
+            options.client_url = baseURL + getMonitorRelativeURL(monitorInfo.id);
+        }
+
+        debug(options);
```

Comment:
Should this be changed to `log.debug()`. `debug()` has been depreceated since #910